app.go

```diff
@@ -179,7 +179,7 @@ func (s *StoreState) SetFeature(name string, val bool) {
 // If an error occurs while trying to update the config the change is rolled back, if the rollback
 // itself fails this function will panic.
 func (s *StoreState) ChangeConfigSetting(name, value string) error {
-	cfg := s.Config()
+	cfg := loadOnChainConfig(s.store)
```

Comment:
If you're reloading the config from the store before changing it then we don't need to bother with backup & rollback since the current config instance isn't modified. But we will need to update the cached config instance after `s.store.Set([]byte(configKey), configBytes)` (or at the very least reset the cached instance to `nil`)

Reply:
Yeah, we do clear config cache after ChangeConfigSetting is called. https://github.com/loomnetwork/loomchain/pull/1439#discussion_r317565193

Reply:
That's in another function, might as well be in another galaxy.

Reply:
No it is in the same galaxy. `ChangeConfigSetting` is only called in `UpdateConfig`